src/platform/plugins/shared/discover/public/application/main/components/layout/cascaded_documents/blocks/use_row_header_components.tsx

```diff
@@ -0,0 +1,523 @@
+        return;
+      }
+
+      this.updateESQLQuery(updatedQuery);
+      return this.closeActionMenu();
+    },
+  },
+  {
+    name: i18n.translate('discover.esql_data_cascade.row.action.view_docs', {
+      defaultMessage: 'Open in new discover tab',
+    }),
+    icon: 'discoverApp',
+    'data-test-subj': 'dscCascadeRowContextActionOpenInNewTab',
+    onClick(this: RowClickActionContext, e) {
+      e.preventDefault();
+
+      return this.openInNewTab({
+        globalState: this.globalState,
+        appState: {
+          query: constructCascadeQuery({
+            query: this.editorQuery,
+            dataView: this.dataView,
+            esqlVariables: this.esqlVariables,
+            nodeType: 'leaf',
+            nodePath: [this.rowContext.groupId],
+            nodePathMap: { [this.rowContext.groupId]: this.rowContext.groupValue },
+          }),
+        },
+      });
+    },
+  },
+];
+
+interface ContextMenuProps
+  extends Pick<
+    RowClickActionContext,
+    | 'editorQuery'
+    | 'globalState'
+    | 'openInNewTab'
+    | 'dataView'
+    | 'esqlVariables'
+    | 'statsFieldSummary'
+    | 'updateESQLQuery'
+  > {
+  row: RowContext;
+  services: UnifiedDataTableProps['services'];
+  close: RowClickActionContext['closeActionMenu'];
+}
+
+const ContextMenu = React.memo(
+  ({
+    row,
+    services,
+    editorQuery,
+    statsFieldSummary,
+    esqlVariables,
+    dataView,
+    close,
+    globalState,
+    openInNewTab,
+    updateESQLQuery,
+  }: ContextMenuProps) => {
+    const rowStatsFieldSummary = useMemo(() => {
+      return statsFieldSummary?.[row.groupId];
+    }, [statsFieldSummary, row.groupId]);
+
+    const groupType = rowStatsFieldSummary
+      ? getStatsGroupFieldType(rowStatsFieldSummary)
+      : undefined;
+
+    const rowDataViewField = useMemo(() => {
+      const fieldParamDef = getFieldParamDefinition(
+        row.groupId,
+        rowStatsFieldSummary?.terminals ?? [],
+        esqlVariables
+      );
+
+      if (typeof fieldParamDef !== 'string') {
+        return undefined;
+      }
+
+      return dataView.fields.getByName(fieldParamDef ?? row.groupId);
+    }, [dataView.fields, esqlVariables, row.groupId, rowStatsFieldSummary?.terminals]);
+
+    const panels = useMemo<EuiContextMenuPanelDescriptor[]>(() => {
+      return [
+        {
+          id: `${row.groupId}-${row.groupValue}-context-menu`,
+          items: contextRowActions.reduce(
+            (acc, { renderFor, enabledWhenFilterable, ...action }) => {
+              if (renderFor && renderFor !== groupType) {
+                return acc;
+              }
+
+              return acc.concat({
+                ...action,
+                disabled:
+                  (enabledWhenFilterable === true &&
+                    rowDataViewField &&
+                    !rowDataViewField?.filterable) ||
+                  !row.groupValue,
+                onClick: (action.onClick as MouseEventHandler<Element>)?.bind({
```

Comment:
```suggestion
                onClick: action.onClick?.bind({
```
Not sure the cast is needed here.